### ISSUE TITLE
Native image broken in 3.40.0.0

### ISSUE THREAD

**Issue by alexeysharandin** [xerial/sqlite-jdbc#816]
In 3.36.0.3-3.39.4.1 native image compilation works fine in 3.40.0.0 got a error.

Job for 3.39.x here
https://github.com/quarkiverse/quarkus-jdbc-sqlite/actions/runs/3571186592/jobs/6002838299

for 3.40.x here
https://github.com/quarkiverse/quarkus-jdbc-sqlite/actions/runs/3537867813/jobs/5938214363#step:6:69896

Error:
```
Failed to load native library:sqlite-3.40.0.0-8c89f691-ab4a-4354-9e28-7d612301bf5e-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: Can't load library: /tmp/sqlite-3.40.0.0-8c89f691-ab4a-4354-9e28-7d612301bf5e-libsqlitejdbc.so
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibrarySupport.loadLibraryAbsolute(NativeLibrarySupport.java:92)
```

**Environment **
 - OS: ubuntu
 - CPU x86
 - sqlite-jdbc version 3.40.0.0
 -  Quarkus 2.9+

**Comment by gotson** [comment]
You will need to do some debugging on your side, as the above doesn't help at all to understand what's going on. Native tests have been green in CI for weeks but actually don't run, and i haven't found a way to fix them yet.

**Comment by alexeysharandin** [comment]
Looks like it's coming from graalVM linkage from here: https://github.com/oracle/graal/blob/44e68777b130c8ee781c72b0e24dfec922712d85/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/NativeLibrarySupport.java#L123
because file.getCanonicalPath() throws IOException for 3.40.0.0 only (after rollback to 3.39.x it's working fine again)

No idea why it's happen

**Comment by pettyalex** [comment]
Oh, this throwing must be why JNI access to Throwable is necessary https://github.com/xerial/sqlite-jdbc/pull/826

@alexeysharandin Does your Quarkus application do runtime or build-time class initialization? With runtime class initialization, sqlite-jdbc is working for me in a Spring Boot native GraalVM image with  #826

**Comment by gotson** [comment]
Will most likely be fixed by #823

**Comment by alexeysharandin** [comment]
@pettyalex nothing added from sqlite-jdbc. Quarkus (GraalVM) trigger the same native-image.properties file

@gotson tried run tests in native locally and get:
```
Failed to load native library:sqlite-3.40.1.0-4bd684cf-3ed0-4ef6-ac15-74082dd240ca-libsqlitejdbc.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: Can't load library: /tmp/sqlite-3.40.1.0-4bd684cf-3ed0-4ef6-ac15-74082dd240ca-libsqlitejdbc.so
        at com.oracle.svm.core.jdk.NativeLibrarySupport.loadLibraryAbsolute(NativeLibrarySupport.java:104)
        at java.lang.ClassLoader.loadLibrary(ClassLoader.java:54)
        at java.lang.Runtime.load0(Runtime.java:755)
        at java.lang.System.load(System.java:1953)
        at org.sqlite.SQLiteJDBCLoader.loadNativeLibrary(SQLiteJDBCLoader.java:280)
```

and 
```
2023-01-09 12:26:44,606 ERROR [io.qua.run.Application] (main) Failed to start application (with profile prod): java.lang.UnsatisfiedLinkError: org.sqlite.core.NativeDB._open_utf8([BI)V [symbol: Java_org_sqlite_core_NativeDB__1open_1utf8 or Java_org_sqlite_core_NativeDB__1open_1utf8___3BI]
        at com.oracle.svm.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:153)
        at com.oracle.svm.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:52)
        at org.sqlite.core.NativeDB._open_utf8(NativeDB.java)
        at org.sqlite.core.NativeDB._open(NativeDB.java:79)
        at org.sqlite.core.DB.open(DB.java:216)
        at org.sqlite.SQLiteConnection.open(SQLiteConnection.java:285)
        at org.sqlite.SQLiteConnection.<init>(SQLiteConnection.java:65)
        at org.sqlite.jdbc3.JDBC3Connection.<init>(JDBC3Connection.java:28)
        at org.sqlite.jdbc4.JDBC4Connection.<init>(JDBC4Connection.java:19)
        at org.sqlite.JDBC.createConnection(JDBC.java:104)
        at org.sqlite.JDBC.connect(JDBC.java:77)
        at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:226)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:535)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:516)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.lang.Thread.run(Thread.java:833)
        at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:704)
        at com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:202)
```

**Comment by alexeysharandin** [comment]
steps to reproduce

1. ```git clone https://github.com/quarkiverse/quarkus-jef.git```
2. in root pom.xml replace
```
<sqlite-jdbc.version>3.39.4.1</sqlite-jdbc.version>
```
to
```
<sqlite-jdbc.version>3.40.1.0-SNAPSHOT</sqlite-jdbc.version>
```
and add
```
<repositories>
    <repository>
      <id>oss.sonatype.org-snapshot</id>
      <url>https://oss.sonatype.org/content/repositories/snapshots</url>
      <releases>
        <enabled>false</enabled>
      </releases>
      <snapshots>
        <enabled>true</enabled>
      </snapshots>
    </repository>
  </repositories>
```
4.  run ```mvn package``` - all test fine in JVM mode
5. run ```mvn package -Pnative-image```
6. execute ```./integration-tests/target/quarkus-jdbc-sqlite-integration-tests-1.0.4-SNAPSHOT-runner``` and get error


BTW: If you use GraalVM less then 22.3 need to downgrade Quarkus version in main pom.xml
to ```<quarkus.version>2.11.0.Final</quarkus.version>``` for example

**Comment by gotson** [comment]
Please try the latest snapshot following the merge of #823

**Comment by alexeysharandin** [comment]
> Please try the latest snapshot following the merge of #823

https://github.com/quarkiverse/quarkus-jdbc-sqlite/ - works fine in native in latest snapshot.

**Comment by gotson** [comment]
🎉 This issue has been resolved in `3.40.1.0` ([Release Notes](https://github.com/xerial/sqlite-jdbc/releases/tag/3.40.1.0))
